DiscordBot.Commands/Helpers/YoutubeSearchHelper.cs

```diff
@@ -0,0 +1,68 @@
+﻿
+using DiscordBot.Commands.Helpers.Models;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using System.Diagnostics;
+
+namespace DiscordBot.Commands.Helpers
+{
+    public interface IYoutubeSearchHelper
+    {
+        YoutubeVideoData GetYoutubeVideoData(string urlOrQuery);
+    }
+
+    public class YoutubeSearchHelper : IYoutubeSearchHelper
+    {
+        private const string HTTP = "http://";
+        private const string HTTPS = "https://";
+        private readonly ILogger<YoutubeSearchHelper> _logger;
+
+        public YoutubeSearchHelper(ILogger<YoutubeSearchHelper> logger) 
+        { 
+            _logger = logger;
+        }
+
+        public YoutubeVideoData GetYoutubeVideoData(string urlOrQuery)
+        {
+            var text = urlOrQuery.ToLower().Trim();
+
+            if(text.StartsWith(HTTP) || text.StartsWith(HTTPS))
+            {
+                return GetVideoDataFromUri(urlOrQuery);
```

Comment:
Also an idea for future: how about we force https by swapping it in the url even if http was provided?

Reply:
Youtube automatically forwards `http` to `https`, and usually all links have `https` in them. Handling `http` is if for some reason url was with this. Also Also it is possible to query based on url to get the same video but this is not consistent. This is why this if statement is even required.